rust/src/mask/config/mod.rs

```diff
@@ -34,24 +41,21 @@ pub enum GroupType {
 }
 
 impl TryFrom<u8> for GroupType {
-    type Error = InvalidMaskConfig;
+    type Error = InvalidMaskConfigError;
 
-    /// Get the group type. Fails if the encoding is unknown.
     fn try_from(byte: u8) -> Result<Self, Self::Error> {
-        use GroupType::{Integer, Power2, Prime};
-
-        Ok(match byte {
-            0 => Integer,
-            1 => Prime,
-            2 => Power2,
-            _ => return Err(InvalidMaskConfig::GroupType),
-        })
+        match byte {
+            0 => Ok(GroupType::Integer),
+            1 => Ok(GroupType::Prime),
+            2 => Ok(GroupType::Power2),
+            _ => Err(InvalidMaskConfigError::GroupType),
+        }
     }
 }
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 #[repr(u8)]
-/// The data type of the numbers to be masked.
+/// The original primitive data type of the numerical values.
```

Comment:
I feel like "numerical values" is a bit too broad. It think "numerical values to mask" would make what this is about clearer.

Reply:
changed.